taker-frontend/src/useEventSource.ts

```diff
@@ -1,15 +1,17 @@
-import { useToast } from "@chakra-ui/react";
 import { useEffect, useRef, useState } from "react";
 
-export function useEventSource(url: string, withCredentials?: boolean) {
+export function useEventSource(url: string, withCredentials?: boolean, setIsConnected) {
     const [source, setSource] = useState<EventSource | null>(null);
-    const [isConnected, setIsConnected] = useState<boolean>(true);
```

Comment:
I think it would be better to keep this state in here and expose it additionally with the `source`, i.e. return 

```ts
{ source: EventSource, isConnected: boolean }
````